rpcs3/Emu/NP/rpcn_client.cpp

```diff
@@ -295,7 +295,12 @@ bool rpcn_client::connect(const std::string& host)
 	connected = true;
 
 	while (!server_info_received && connected && !is_abort())
-		thread_ctrl::wait_for(5000);
+	{
+		if (in_config)
+			std::this_thread::sleep_for(5ms);
+		else
+			thread_ctrl::wait_for(5000);
```

Comment:
Maybe modify thread_ctrl::_wait_for instead  so in such case it'd wait on a dummy atomic variable with specified timeout.

Reply:
I mean, there's little sense in not allowing thread_ctrl::get_current() == nullptr for it in the first place.

Reply:
Doesn't matter, there is a var for it already and in the future rpcn_client should run completely async from np_handler and go back to sleep_for anyway :).

Reply:
Waiting on dummy atomic would work in case alert = false, but it's true by default. Pretty dangerous change.
Also please remember that all such sleep/wait functions with timeout are undesireable by nature: there shall always be a proper condition to wait for.